handle/worker.go

```diff
@@ -107,22 +111,60 @@ func slave(slave_num int) {
 					break
 				}
 			}
+			ReturnQ <- FinishedTask{200,taskData.UUID,taskData.Url,data}
 		}
 	}
 }
 
 func downloadImage(downloadUrl string) ([]byte, error) {
-	return nil, nil
+	helper.Log.Info(fmt.Sprintf("Start to download %s\n", downloadUrl))
+
+	httpClient := &http.Client{Timeout: time.Second * 5}
+	resp, err := httpClient.Get(downloadUrl)
+
+	if err != nil {
+		helper.Log.Error("Download failed!", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	//check header
+	if resp.StatusCode != 200 {
+		helper.Log.Info("Request is not 200")
+		return nil, ErrDownloadCode
+	}
+
+	mimeType := resp.Header.Get("Content-Type")
+
+	if strings.Contains(mimeType, "image") == false {
+		if ok, _ := regexp.MatchString("(jpeg|jpg|png|gif|bmp|webp|tiff)", downloadUrl); ok == false {
+			helper.Log.Info(fmt.Sprintf("MIME TYPE is %s not an image\n", mimeType))
+			return nil, StatusUnsupportedMediaType //415
+		}
+	}
+	contentLength := resp.Header.Get("Content-Length")
+	if len, _ := strconv.Atoi(contentLength); len > (20 << 20) {
```

Comment:
Involving numerical values, especially the left shift number is defined with const